client-js/app/elements/labrad-registry.ts

```diff
@@ -678,49 +707,44 @@ export class LabradRegistry extends polymer.Base {
    */
   async doDragOp() {
     const newName = this.$.dragNameInput.value;
-    const newPath = JSON.parse(this.$.dragPathInput.value);
-    const oldPath = JSON.parse(this.$.originPath.textContent);
     const oldName = this.$.originName.textContent;
+    const dragType = (this.$.dragOp.innerText === 'Copy') ? 'copy' : 'move';
 
-    if (this.$.dragOp.innerText === 'Copy') {
-      try {
-        this.$.pendingDialog.open();
-        this.$.pendingOp.innerText = 'Copying...';
-        switch (this.$.dragDialog.dragData['kind']) {
-          case 'dir':
+    try {
+      const oldPath = JSON.parse(this.$.originPath.textContent);
+      const newPath = JSON.parse(this.$.dragPathInput.value);
+
+      this.$.pendingOp.innerText = (dragType === "copy") ? 'Copying...' : 'Moving...';
+      this.$.pendingDialog.open();
+
+      switch (this.$.dragDialog.dragData['kind']) {
+        case 'dir':
+          if (dragType === "copy") {
             await this.socket.copyDir({path: oldPath, dir: oldName, newPath: newPath, newDir: newName});
-            break;
+          } else {
+            await this.socket.moveDir({path: oldPath, dir: oldName, newPath: newPath, newDir: newName});
+          }
+          break;
 
-          case 'key':
+        case 'key':
+          if (dragType === "copy") {
             await this.socket.copy({path: oldPath, key: oldName, newPath: newPath, newKey: newName});
-            break;
-        }
-      } catch (error) {
-        this.handleError(error);
-      } finally {
-        this.$.pendingDialog.close();
-        this.$.toastCopySuccess.show();
+          } else {
+            await this.socket.move({path: oldPath, key: oldName, newPath: newPath, newKey: newName});
+          }
+          break;
       }
-    } else if (this.$.dragOp.innerText === 'Move') {
-      try {
-        this.$.pendingDialog.open();
-        this.$.pendingOp.innerText = 'Moving...';
-        switch (this.$.dragDialog.dragData['kind']) {
-          case 'dir':
-            await this.socket.moveDir({path: oldPath, dir: oldName, newPath: newPath, newDir: newName});
-            break;
 
-          case 'key':
-            await this.socket.move({path: oldPath, key: oldName, newPath: newPath, newKey: newName});
-            break;
-        }
-      } catch (error) {
-        this.handleError(error);
-      } finally {
-        this.$.pendingDialog.close();
+      if (dragType === "copy") {
+        this.$.toastCopySuccess.show();
+      } else {
         this.$.toastMoveSuccess.show();
       }
```

Comment:
I'm ok with killing the pending dialog. Probably best would be to show a spinner on the dialog you've just used to trigger an operation. Then if the operation fails, we show an error and the user can try again. If it succeeds, the dialog just closes, no more UI changes needed.